Sources/Helpers/OAGPXUIHelper.mm

```diff
@@ -116,6 +116,21 @@ + (NSString *) getDescription:(OAGPX *)gpx
     return [NSString stringWithFormat:@"%@ • %@", dist, wpts];
 }
 
++ (NSDate *) getCreationDate:(NSString *)filePath 
+{
+    OAGPXDocument *gpxFile = [[OAGPXDocument alloc] initWithGpxFile:filePath];
```

Comment:
This will cause heavy load, especially while using in sorting. Just drop this method and change it with something optimal. Looks like `creationDate` should be in `OAGPX`. Better to check java code.